Numsense/German.fs

```diff
@@ -0,0 +1,165 @@
+    | _ -> ""
+
+let convToInt (str : string) =
+    Convert.ToInt32(str)
+
+let combineParts (newPart : string) (result : string) =
+    match (newPart <> "") with
+    | true ->
+        match (result <> "") with
+        | true -> sprintf "%s-%s" newPart result
+        | false -> newPart
+    | false -> result
+
+let getPartOfNumber start length number =
+    let divisor = pown 10 (start - 1)
+    let parts = pown 10 length
+
+    (number / divisor) % parts
+
+let internal toGermanNumber (x : int ) =
+
+    let minus = (x < 0)
+    let absValue = Math.Abs(x)
+
+    let milliarde = absValue |> getPartOfNumber 10 1  |> convertPart ""
+    let million = absValue |> getPartOfNumber 7 3 |> convertPart ""
+    let tausend = absValue |> getPartOfNumber 4 3 |> convertPart ""
+    let rest = absValue |> getPartOfNumber 1 3 |> convertPart "s"
+
+    let milliardeString = match milliarde with
+                          | "" -> ""
+                          | "ein" -> "eine-milliarde"
+                          | _ -> sprintf "%s%s" milliarde "-milliarden"
+    let millionString = match million with
+                        | "" -> ""
+                        | "ein" -> "eine-million"
+                        | _ -> sprintf "%s%s" million "-millionen"
+
+    let tausendString = match tausend with
+                        | "" -> ""
+                        | _ -> sprintf "%s%s" tausend "-tausend"
+
+    let result = rest
+                 |> combineParts tausendString
+                 |> combineParts millionString
+                 |> combineParts milliardeString
+
+    match minus with
+    | true -> sprintf "minus-%s" result
+    | false -> result
+
+let internal toGermanImp (x : int)  =
+    match x with
+    | 0 -> "null"
+    | _ -> toGermanNumber x
+
+let internal tryParseGermanImp (x : string) =
+    let rec conv acc (candidate : string) =
+        let trimmed = candidate.Trim()
+        match trimmed with
+        | ""                      -> Some acc
+        | StartsWith "-"         t
+        | StartsWith "UND"       t -> conv                acc  t
+        | StartsWith "NULL"      t -> conv          (0  + acc) t
+        | StartsWith "EINE"      t -> conv          (1  + acc) t
+        | StartsWith "EINS"      t -> conv          (1  + acc) t
+        | StartsWith "EIN"       t -> conv          (1  + acc) t
+        | StartsWith "ZWEI"      t -> conv          (2  + acc) t
+        | StartsWith "ZEHN"      t -> conv         (10  + acc) t
+        | StartsWith "ELF"       t -> conv         (11  + acc) t
+        | StartsWith "ZWÖLF"     t -> conv         (12  + acc) t
+        | StartsWith "DREIZEHN"  t -> conv         (13  + acc) t
+        | StartsWith "FÜNFZEHN"  t -> conv         (15  + acc) t
+        | StartsWith "SECHZEHN"  t -> conv         (16  + acc) t
+        | StartsWith "SIEBZEHN"  t -> conv         (17  + acc) t
+        | StartsWith "EHN"       t // matches 'een' in 'eighteen'
```

Comment:
18 is "achtzehn" in German, consisting of "acht" for 8 and "zehn" for 10, so this seems to be just a copy-and-paste "remnant" from English. I can't think of any case this would match.